pkg/logger/logger.go

```diff
@@ -14,39 +14,169 @@
+	defaultLogger.Trace(ctx, msg, args...)
+}
+
+// Debug logs at LevelDebug with the given context.
+func Debug(ctx context.Context, msg string, args ...any) {
+	defaultLogger.Debug(ctx, msg, args...)
+}
+
+// Info logs at LevelInfo with the given context.
+func Info(ctx context.Context, msg string, args ...any) {
+	defaultLogger.Info(ctx, msg, args...)
+}
+
+// Warn logs at LevelWarn with the given context.
+func Warn(ctx context.Context, msg string, args ...any) {
+	defaultLogger.Warn(ctx, msg, args...)
+}
+
+// Error logs at LevelError with the given context.
+func Error(ctx context.Context, msg string, args ...any) {
+	defaultLogger.Error(ctx, msg, args...)
+}
+
+type contextualLogger struct {
+	logger *slog.Logger
+	depth  *int
+}
+
+// With returns a Logger that includes the given attributes in each output operation.
+func (l *contextualLogger) With(args ...any) Logger {
+	return &contextualLogger{logger: l.logger.With(args...), depth: l.depth}
+}
+
+// WithGroup returns a Logger that starts a group
+func (l *contextualLogger) WithGroup(name string) Logger {
+	return &contextualLogger{logger: l.logger.WithGroup(name), depth: l.depth}
+}
+
+// Trace logs at LevelInfo with the given context.
+func (l *contextualLogger) Trace(ctx context.Context, msg string, args ...any) {
+	l.Log(ctx, LevelTrace, msg, args...)
+}
+
+// Debug logs at LevelInfo with the given context.
+func (l *contextualLogger) Debug(ctx context.Context, msg string, args ...any) {
+	l.Log(ctx, slog.LevelDebug, msg, args...)
+}
+
+// Info logs at LevelInfo with the given context.
+func (l *contextualLogger) Info(ctx context.Context, msg string, args ...any) {
+	l.Log(ctx, slog.LevelInfo, msg, args...)
+}
+
+// Warn logs at LevelInfo with the given context.
+func (l *contextualLogger) Warn(ctx context.Context, msg string, args ...any) {
+	l.Log(ctx, slog.LevelWarn, msg, args...)
+}
+
+// Error logs at LevelInfo with the given context.
+func (l *contextualLogger) Error(ctx context.Context, msg string, args ...any) {
+	l.Log(ctx, slog.LevelError, msg, args...)
+}
+
+// Log logs at Level with the given context.
+func (l *contextualLogger) Log(ctx context.Context, level slog.Level, msg string, args ...any) {
+	ctx = context.WithValue(ctx, depthCtxKey, l.depth)
+	l.logger.Log(ctx, level, msg, args...)
+}
+
+// New make a new Logger
+func New(h slog.Handler) Logger {
+	return &contextualLogger{logger: slog.New(h)}
+}
+
+// WithAttr 添加自定义属性
+func WithAttr(ctx context.Context, attrs ...slog.Attr) context.Context {
+	rawAttrs, ok := ctx.Value(attrCtxKey).([]slog.Attr)
 	if !ok {
-		return a
+		return context.WithValue(ctx, attrCtxKey, attrs)
+	}
+
+	rawAttrs = append(rawAttrs, attrs...)
+	ctx = context.WithValue(ctx, attrCtxKey, rawAttrs)
+	return ctx
+}
+
+// RidAttr request_id类型Attr
+func RidAttr(rid string) slog.Attr {
+	return slog.String("rid", rid)
+}
+
+// ErrAttr 错误类型Attr
+func ErrAttr(err error) slog.Attr {
+	return slog.String("err", err.Error())
+}
```

Comment:
这个会是一个巨高频的用法，看下要不要简写成，或者增加一个直接带Err入参的log方法


```suggestion
// E 错误类型Attr
func E(err error) slog.Attr {
	return slog.String("err", err.Error())
}
```